TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,2695 @@
+        protected virtual void LeavePerformProcedure(PerformProcedure node)
+        {
+        }
+
+        /// <summary>
+        /// Enter a Declarative
+        /// </summary>
+        /// <param name="node">The Declarative node</param>
+        protected virtual void EnterDeclaratives(Declaratives node)
+        {
+            System.Diagnostics.Debug.Assert(this.CurrentProgramCfgBuilder.CurrentDeclarativesContext == null);
+            this.CurrentProgramCfgBuilder.CurrentDeclarativesContext = new DeclarativesContext(this.CurrentProgramCfgBuilder);
+            this.CurrentProgramCfgBuilder.CurrentDeclarativesContext.Start(this.CurrentProgramCfgBuilder.CurrentBasicBlock);
+        }
+
+        /// <summary>
+        /// Leave a Declarative
+        /// </summary>
+        /// <param name="node">The Declarative node</param>
+        protected virtual void LeaveDeclaratives(Declaratives node)
+        {
+            System.Diagnostics.Debug.Assert(this.CurrentProgramCfgBuilder.CurrentDeclarativesContext != null);
+
+            //The next block.
+            var nextBlock = this.CurrentProgramCfgBuilder.CreateBlock(null, true);
+            this.CurrentProgramCfgBuilder.CurrentDeclarativesContext.End(nextBlock);
+            this.CurrentProgramCfgBuilder.CurrentBasicBlock = nextBlock;
+
+            this.CurrentProgramCfgBuilder.CurrentDeclarativesContext = null;
+        }
+
+        /// <summary>
+        /// Enter any Statement
+        /// </summary>
+        /// <param name="node">The Statement node to be entered</param>
+        protected virtual void EnterStatement(Node node)
+        {
+            AddCurrentBlockNode(node);
+        }
+
+        /// <summary>
+        /// Leave any Statement
+        /// </summary>
+        /// <param name="node">The Statement node to be leaves</param>
+        protected virtual void LeaveStatement(Node node)
+        {
+
+        }
+
+        /// <summary>
+        /// Add a Node to the current block.
+        /// </summary>
+        /// <param name="node">The node to be added</param>
+        protected virtual void AddCurrentBlockNode(Node node)
+        {
+            System.Diagnostics.Debug.Assert(this.CurrentProgramCfgBuilder.CurrentBasicBlock != null);
+            if (this.CurrentProgramCfgBuilder.CurrentBasicBlock != null)
+            {
+                this.CurrentProgramCfgBuilder.CurrentBasicBlock.Instructions.AddLast(node);
+                this.CurrentProgramCfgBuilder.Cfg.BlockFor[node] = this.CurrentProgramCfgBuilder.CurrentBasicBlock;
+            }
+        }
+
+        /// <summary>
+        /// Create a basic block for a node.
+        /// </summary>
+        /// <param name="node">The leading node of the block</param>
+        /// <param name="addToCurrentSentence">true if the block must be added to the current Sentence, false otherwise.</param>
+        /// <returns>The new Block</returns>
+        internal BasicBlockForNode CreateBlock(Node node, bool addToCurrentSentence)
```

Comment:
Second parameter can be removed, set to true in every usages.

Reply:
hum...

Reply:
After PR #1738, the parameter is now used once.